src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -184,6 +184,26 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
                     await RefreshKeyValueCollections(cancellationToken).ConfigureAwait(false);
                     await RefreshKeyValueAdapters(cancellationToken).ConfigureAwait(false);
                 }
+                catch (Exception exception) when (exception is RequestFailedException ||

```

Comment:
I was expecting that we just move the cache expiration update to before we make a server call. Does that not work?

Reply:
Network delay shouldn't be an objective for Refresh. It's meant to be precise anyway.

Reply:
Okay, updating cache expiration time before server call.

> We probably can do it in `finally` instead.

Just realized that if we do it in catch or finally block, we wont be able to control the exact sentinel key for which cache expiration needs to be updated. The new approach solves both issues.